lnwallet/payment_descriptor.go

```diff
@@ -0,0 +1,224 @@
+	// into their update log.
+	Add updateType = iota
+
+	// Fail is an update type which removes a prior HTLC entry from the
+	// log. Adding a Fail entry to one's log will modify the _remote_
+	// party's update log once a new commitment view has been evaluated
+	// which contains the Fail entry.
+	Fail
+
+	// MalformedFail is an update type which removes a prior HTLC entry
+	// from the log. Adding a MalformedFail entry to one's log will modify
+	// the _remote_ party's update log once a new commitment view has been
+	// evaluated which contains the MalformedFail entry. The difference
+	// from Fail type lie in the different data we have to store.
+	MalformedFail
+
+	// Settle is an update type which settles a prior HTLC crediting the
+	// balance of the receiving node. Adding a Settle entry to a log will
+	// result in the settle entry being removed on the log as well as the
+	// original add entry from the remote party's log after the next state
+	// transition.
+	Settle
+
+	// FeeUpdate is an update type sent by the channel initiator that
+	// updates the fee rate used when signing the commitment transaction.
+	FeeUpdate
+)
+
+// String returns a human readable string that uniquely identifies the target
+// update type.
+func (u updateType) String() string {
+	switch u {
+	case Add:
+		return "Add"
+	case Fail:
+		return "Fail"
+	case MalformedFail:
+		return "MalformedFail"
+	case Settle:
+		return "Settle"
+	case FeeUpdate:
+		return "FeeUpdate"
+	default:
+		return "<unknown type>"
+	}
+}
+
+// PaymentDescriptor represents a commitment state update which either adds,
+// settles, or removes an HTLC. PaymentDescriptors encapsulate all necessary
+// metadata w.r.t to an HTLC, and additional data pairing a settle message to
+// the original added HTLC.
+//
+// TODO(roasbeef): LogEntry interface??
+//   - need to separate attrs for cancel/add/settle/feeupdate
+type PaymentDescriptor struct {
+	// RHash is the payment hash for this HTLC. The HTLC can be settled iff
+	// the preimage to this hash is presented.
+	RHash PaymentHash
+
+	// RPreimage is the preimage that settles the HTLC pointed to within the
+	// log by the ParentIndex.
+	RPreimage PaymentHash
+
+	// Timeout is the absolute timeout in blocks, after which this HTLC
+	// expires.
+	Timeout uint32
+
+	// Amount is the HTLC amount in milli-satoshis.
+	Amount lnwire.MilliSatoshi
+
+	// LogIndex is the log entry number that his HTLC update has within the
+	// log. Depending on if IsIncoming is true, this is either an entry the
+	// remote party added, or one that we added locally.
+	LogIndex uint64
+
+	// HtlcIndex is the index within the main update log for this HTLC.
+	// Entries within the log of type Add will have this field populated,
+	// as other entries will point to the entry via this counter.
+	//
+	// NOTE: This field will only be populate if EntryType is Add.
+	HtlcIndex uint64
+
+	// ParentIndex is the HTLC index of the entry that this update settles
+	// or times out.
```

Comment:
nit: it would've been slightly easier to review (and verify this is a pure code move), if the linter issues were addressed in a separate commit (non-blocking, just for the next PR).